### PR TITLE
Use tables to drive E2E tests

### PR DESCRIPTION
**Pull request by mem**
A number of E2E suites have this structure:

	return func(t *testing.T) {
		// some thing
		t.Run("testSomething", c.testSomething)
		// another thing
		t.Run("testAnotherThing", c.testAnotherThing)
		// one more
		t.Run("testOneMoreThing", c.testOneMoreThing)
	}

CamelCase is not really necessary here, and it makes things a little
harder to read. Replace those by short sentences that hopefully describe
the group of tests represented by the corresponding function.

The sequence of calls to t.Run is a little repetitive. Replace that by a
map with the test name and the test function and a new helper function
to call the functions.

The result looks like this:

	return testhelper.TestRunner(map[string]func(*testing.T){
		"another thing":  c.testAnotherThing, // another thing
		"one more thing": c.testOneMoreThing, // one more
		"something":      c.testSomething,    // some thing
	})

The tests are sorted alphabetically by their description, so some care
should be paid to NOT create tests that must run in sequence.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

**Description of the Pull Request (PR):**

Write your description of the PR here. Be sure to include as much background,
and details necessary for the reviewers to understand exactly what this is
fixing or enhancing.

**This fixes or addresses the following GitHub issues:**

- Fixes #

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: e2e/internal/testhelper/doc.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// Package testhelper contains a collecton of test helper functions that
```
Comment (sashayakovtseva) [inline e2e/internal/testhelper/doc.go]:
```suggestion
// Package testhelper contains a collection of test helper functions that
```

File: e2e/docker/docker.go [hunk 2]
```diff
@@ -375,12 +376,12 @@ func E2ETests(env e2e.TestEnv) func(*testing.T) {
...
-		t.Run("testDockerDefFile", c.testDockerDefFile)
-		t.Run("testDockerRegistry", c.testDockerRegistry)
-	}
+	return testhelper.TestRunner(map[string]func(*testing.T){
+		"AUFS":             c.testDockerAUFS,
+		"def file":         c.testDockerDefFile,
+		"permissions":      c.testDockerPermissions,
+		"pulls":            c.testDockerPulls,
+		"registry":         c.testDockerRegistry,
+		"whiteout symlink": c.testDockerWhiteoutSymlink,
```
Comment (jscook2345) [inline e2e/docker/docker.go:385]: I know this isn't changed, but what is a `Whiteout Symlink`? Should that be `Without Symlink`? If so, could an issue be opened to fix that?

File: e2e/internal/testhelper/doc.go (hunk 1 above)
Comment (jscook2345) [reply e2e/internal/testhelper/doc.go]: This needs to be fixed to pass the lint tests

**Comment by jscook2345** [comment]
LGTM, assuming everything works once the lint tests pass.

File: e2e/docker/docker.go (hunk 2 above)
Comment (mem) [reply e2e/docker/docker.go:385]: ah, that might be it.

I was wondering about the exact same thing... let me check the test...

Comment (mem) [reply e2e/docker/docker.go:385]: the test doesn't help to understand this as it refers a specific image in docker, and refers to #1592, #1576

It seems they mean "wipeout"? It looks like there are talking about replacing an existing symlink with an actual file in the context of an overlay.

File: e2e/internal/testhelper/doc.go (hunk 1 above)
Comment (mem) [reply e2e/internal/testhelper/doc.go]: thanks!

**Review by WestleyK — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**